portal-impl/src/com/liferay/portal/language/LanguageImpl.java

```diff
@@ -1632,8 +1634,10 @@ private static CompanyLocalesBag _getCompanyLocalesBag(long companyId) {
 		catch (Exception e) {
 		}
 
-		HashMap<String, Locale> groupLanguageCodeLocalesMap = new HashMap<>();
-		HashMap<String, Locale> groupLanguageIdLocalesMap = new HashMap<>();
+		LinkedHashMap<String, Locale> groupLanguageCodeLocalesMap =
```

Comment:
@Seiphon is this really needed? I only see later the groupLanguageIdLocalesMap is used for returning all values, which means order could matter. For groupLanguageCodeLocalesMap, we never use all its values, only for single key checking.

Reply:
Hi @shuyangzhou 
You are right, groupLanguageCodeLocalesMap doesn't order matter. I changed it only because groupLanguageCodeLocalesMap and groupLanguageIdLocalesMap are related, and I want them have consistent data type. Since it's not necessary, I will change it back and resubmit the pull request.
Thanks,
Seiphon